docs/okeps/okep-5494-ovn-kubernetes-mcp-server.md

```diff
@@ -0,0 +1,740 @@
+  extra context.
+* Investigate if there is potential for converting it also into a remediation
+  system (this would need write access)
+* Work with the Layered project community teams (OVN, OpenvSwitch, Kernel)
+  for each of those layers to also own an MCP server since they know their
+  stack better. So instead of 1 OVN-Kubernetes MCP Server it would be a
+  bunch of servers each owned by specific upstream community projects for
+  better maintenance and ownership
+* The same MCP Server could also potentially be used as a
+  "OVN-Kubernetes network plugin - know it better" chatbot but that is not
+  the initial goal here.
+
+# Non-Goals
+
+* Phase3 includes this getting productized and shipped to end-users using
+  OVN-Kubernetes in some far fetched future to run it on production to
+  troubleshoot. But this would require:
+    * Having a better overall architecture for the wholesome agentic AI
+      troubleshooting solution on a cluster
+    * Solving the compute problem of how and where to run the model
+    * Having an air tight security vetting.
+
+  So running this stack on a live production environment has been marked as
+  non-goal.
+
+* We will not be solving the problem around which LLM should be used. Most of
+  the good ones (based on shift week demo) were proprietary (claude sonnet4
+  and gemini2.5pro) but testing all LLMs and coming up with which works the
+  best is not in scope
+    * By not solving this problem as part of this enhancement, we risk having
+      to deal with "long-context windows causing hallucinations" but that's
+      where RAGing mentioned in future goals could help.
+* We will also not be developing our own model to teach and maintain it
+    * By not solving this problem, we also risk relying on proprietary models
+      knowing and learning what we want them to learn but having no control on
+      how fast they learn it. Again RAGing could help here.
+    * So the quality here will heavily depend on how good the brain LLM is
+      which basically won't be in our control much.
+
+# Introduction
+
+An engineer troubleshooting OVN-Kubernetes usually uses the following set of
+CLI tools in a layered fashion:
+
+* Kubernetes and OVN-Kubernetes layer - this cluster state information is
+  gathered as part of must-gather for offline troubleshooting
+    * **kubectl** commands like list, get, describe, logs, exec, events to know
+      everything about the Kubernetes API state of the feature and to know what
+      the ovnkube pods were doing through the logs they generate during that
+      window
+    * **ovnkube-trace** which executes ovn-trace and ovs ofproto trace and
+      detrace commands (this tool doesn't support all scenarios - it's not
+      maintained well)
+    * Future-tool - ovnkube CLI (need to ask NVIDIA what's the status here)
+        * A place for K8s state-database syncer should/could potentially exist
+* OVN Layer - RHEL - OVN databases are gathered as part of must-gather for
```

Comment:
Lets drop `RHEL` and any other redhat/ocp references?

Reply:
done. good point
I can see I did a lousy job of copying enhancement from downstream to upstream :)